src/pudl/metadata/classes.py

```diff
@@ -2028,6 +2045,48 @@ def sort_resource_names(resource: Resource):
 
         return sorted(resources, key=sort_resource_names, reverse=False)
 
+    @cached_property
+    def encoders(self) -> dict[SnakeCase, Encoder]:
+        """Compile a mapping of field names to their encoders, if they exist.
+
+        This dictionary will be used many times, so it makes sense to build it once
+        when the Package is instantiated so it can be reused.
+        """
+        encoded_fields = [
+            field
+            for res in self.resources
+            for field in res.schema.fields
+            if field.encoder
+        ]
+        encoders: dict[SnakeCase, Encoder] = {}
+        for field in encoded_fields:
+            if field.name not in encoders:
+                encoders[field.name] = field.encoder
+            else:
+                # We have a field which shows up in multiple tables, and we want to
+                # verify that the encoders are the same across all of them.
+                pd.testing.assert_frame_equal(encoders[field.name].df, field.encoder.df)
+                assert encoders[field.name].code_fixes == field.encoder.code_fixes
+                assert encoders[field.name].ignored_codes == field.encoder.ignored_codes
+        return encoders
+
+    def encode(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Clean up all coded columns in a dataframe based on PUDL coding tables.
+
+        Returns:
+            A modified copy of the input dataframe.
+        """
+        encoded_df = df.copy()
+        for col in encoded_df.columns:
+            if col in self.encoders:
+                encoded_df[col] = self.encoders[col].encode(
+                    encoded_df[col], dtype=Field.from_id(col).to_pandas_dtype()
+                )
+        return encoded_df
+
+
+PUDL_PACKAGE = Package.from_resource_ids()
```

Comment:
This has to happen after the definition of the `Package` class, and I did it here rather in `pudl/metadata/__init__.py` because it's used below a couple of times, but I import it at the metadata package level to keep the path shorter and kind of unbury it from this middle of the module definition.

Reply:
maybe add this as a comment so someone later doesn't try to move it and get all janked up with imports